### PR TITLE
Update notebooks tests

### PR DESCRIPTION
**Pull request by ps48**
### Description

Update notebooks tests, this PR depends on notebooks updates to be merged by this PR: https://github.com/opensearch-project/dashboards-observability/pull/1367

### Issues Resolved

https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/897

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ps48** [comment]
## Local run results


https://github.com/opensearch-project/opensearch-dashboards-functional-test/assets/4348487/9bb3914e-eeb1-4e1c-b4e1-b6f7bbc50b1e



```
========================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        9.5.4                                                                          │
  │ Browser:        Electron 94 (headless)                                                         │
  │ Node Version:   v18.16.0 (/Users/sgguruda/.nvm/versions/node/v18.16.0/bin/node)                │
  │ Specs:          1 found (plugins/observability-dashboards/6_notebooks.spec.js)                 │
  │ Searched:       cypress/integration/plugins/observability-dashboards/6_notebooks.spec.js       │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  plugins/observability-dashboards/6_notebooks.spec.js                            (1 of 1)
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Couldn't determine Mocha version


  Testing notebook actions
    ✓ Creates a code paragraph (19254ms)
    ✓ Renders markdown (12740ms)
    ✓ Adds a SQL query paragraph (6234ms)

  Test reporting integration if plugin installed
    ✓ Create in-context PDF report from notebook
    ✓ Create in-context PNG report from notebook (5103ms)
    ✓ Create on-demand report definition from context menu (11148ms)
    ✓ View reports homepage from context menu (5293ms)

  clean up all test data
    ✓ Cleans up test notebooks (5908ms)


  8 passing (1m)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        8                                                                                │
  │ Passing:      8                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     1 minute, 10 seconds                                                             │
  │ Spec Ran:     plugins/observability-dashboards/6_notebooks.spec.js                             │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Video)

  -  Started processing:  Compressing to 32 CRF
    Compression progress:  4%
    Compression progress:  12%
  -  Finished processing: /Users/sgguruda/work/git_codebase/temp/opensearch-dashboard   (28 seconds)
                          s-functional-test/cypress/videos/plugins/observability-dash
                          boards/6_notebooks.spec.js.mp4


========================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  plugins/observability-dashboards/6_      01:10        8        8        -        -        - │
  │    notebooks.spec.js                                                                           │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        01:10        8        8        -        -        -

Done in 117.75s.
```

**Review by joshuali925 — APPROVED** [review]

**Comment by ps48** [comment]
@Hailong-am, @wanglam Can you please review this PR? The dependent PR in dashboards-observability is merged in.

File: cypress/integration/plugins/observability-dashboards/6_notebooks.spec.js [hunk 1]
```diff
@@ -6,157 +6,170 @@
...
+  cy.visit(`${BASE_PATH}/app/observability-notebooks#/`);
+};
+
 const moveToTestNotebook = () => {
   cy.visit(`${BASE_PATH}/app/observability-notebooks#/`, {
     timeout: delayTime * 3,
   });
+
+  // Reload page to load notebooks if they are not flushed in OpenSearch index yet.
+  cy.reload();
```
Comment (wanglam) [inline cypress/integration/plugins/observability-dashboards/6_notebooks.spec.js:29]: Shall we check if they are already flushed?

File: cypress/integration/plugins/observability-dashboards/6_notebooks.spec.js [hunk 2]
```diff
@@ -6,157 +6,170 @@
...
-    cy.get('.euiTextAlign').contains('No paragraphs').should('exist');
+    cy.location('pathname', { timeout: delayTime * 3 }).should(
+      'include',
+      '/reports-dashboards'
+    );
   });
+});
 
+describe('clean up all test data', () => {
   it('Cleans up test notebooks', () => {
```
Comment (wanglam) [inline cypress/integration/plugins/observability-dashboards/6_notebooks.spec.js:153]: Nit: Shall we record the created notebook ids and just clear these items?

**Review by Hailong-am — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Comment by ruanyl** [comment]
@ps48 Added `backport 2.x` label

**Comment by ruanyl** [comment]
@ps48 The backport to `2.x` is failed, could you do a manual backport?

**Comment by ps48** [comment]
Hi @ruanyl & @wanglam, can you please review the backport PR here: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1021 Thanks!

**Pull request merged**